mavros/src/plugins/setpoint_position.cpp

```diff
@@ -93,12 +100,25 @@ class SetpointPositionPlugin : public plugin::PluginBase,
 		 */
 		const uint16_t ignore_all_except_xyz_y = (1 << 11) | (7 << 6) | (7 << 3);
 
-		auto p = ftf::transform_frame_enu_ned(Eigen::Vector3d(tr.translation()));
-		auto q = ftf::transform_orientation_enu_ned(
-					ftf::transform_orientation_baselink_aircraft(Eigen::Quaterniond(tr.rotation())));
+		auto p = [&] {
+			if (static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_NED || static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_OFFSET_NED) {
```

Comment:
For `BODY_OFFSET_NED` we should have a way of setting the offset. That would of course require adding some new functions to `frame_tf.h`. But guess we can spare that work since the idea in the near future is to change everything from static transforms to using `tf2`.